contracts/TridentRouter.sol

```diff
@@ -68,16 +70,15 @@ contract TridentRouter is ITridentRouter, SelfPermit, Multicall {
     /// @param params This includes the addresses of the tokens, pools, amount of token A to swap,
     /// minimum amount of token B after the swap and data required by the pools for the swaps.
     /// @dev Ensure that the pools are trusted before calling this function. The pools can steal users' tokens.
-    function exactInput(ExactInputParams calldata params) public payable returns (uint256 amountOut) {
+    function exactInput(ExactInputParams calldata params) external payable returns (uint256 amountOut) {
         // Pay the first pool directly.
         bento.transfer(params.tokenIn, msg.sender, params.path[0].pool, params.amountIn);
         // Call every pool in the path.
         // Pool `N` should transfer its output tokens to pool `N+1` directly.
         // The last pool should transfer its output tokens to the user.
         // If the user wants to unwrap `wETH`, the final destination should be this contract and
         // a batch call should be made to `unwrapWETH`.
-        uint256 n = params.path.length;
```

Comment:
Is this not an optimisation? I didn't run the numbers but afaik it is ([caching-the-length-in-for-loops](https://gist.github.com/hrkrshnn/ee8fabd532058307229d65dcd5836ddc#caching-the-length-in-for-loops))

Reply:
@z0r0z

Reply:
@gasperbr I just did some testing and including cache increases gas cost 
![image](https://user-images.githubusercontent.com/92001561/166683333-9511c455-e448-42cc-a208-3c3fb8ffb57e.png)

gist, https://gist.github.com/z0r0z/4187d4f39b9a2a6e89941f9f8e86682b

Reply:
let me know if this makes sense. I see explanation of how optimizer might be working to cache outside of loop here, https://gist.github.com/hrkrshnn/a1165fc31cbbf1fae9f271c73830fdda